src/cadence/testnet/getNFTs.cdc

```diff
@@ -811,6 +813,33 @@ pub fun getStarlyCard(owner: PublicAccount, id: UInt64): NFTData? {
     let nft = col!.borrowStarlyCard(id: id)
     if nft == nil { return nil }
 
+    let viewResolver = col!.borrowViewResolver(id: id)
+    let view = viewResolver.resolveView(Type<StarlyMetadataViews.CardEdition>())
```

Comment:
Hey @vleushin sorry for the delayed response here - It would be great to use the standard metadata view (MetadataViews.Display) for the fields that it supports (name, description and thumbnail). We are in the process of defining generalized standard for metadata too and would love for you to work with Peter and Albert from Flow on this.
Tagging them to chime in: @albeethekid @psiemens 


let viewResolver = col!.borrowViewResolver(id: id)
let displayView = viewResolver.resolveView(Type<MetadataViews.Display>())
let starlyView = viewResolver.resolveView(Type<StarlyMetadataViews.CardEdition>())

if let displayMetadata = view as! MetadataViews.Display? {
    if let starlyMetadata = view as! StarlyMetadataViews.CardEdition? {
        return NFTData(
            contract: contract,
            id: nft!.id,
            uuid: nft!.uuid,
            title: displayMetadata.title,
            description: displayMetadata.description,
            external_domain_view_url: starlyMetadata.url,
            token_uri: nil,
            media: [
                NFTMedia(
                    uri: starlyMetadata.card.mediaSizes[0].url, 
                    mimetype: starlyMetadata.card.mediaType
                )
            ],
            metadata: {
                "id": nft!.starlyID,
                "rarity": starlyMetadata.card.rarity,
                "collectionID": starlyMetadata.collection.id,
                "collectionTitle": starlyMetadata.collection.title,
                "cardID": starlyMetadata.card.id,
                "edition": starlyMetadata.edition,
                "editions": starlyMetadata.card.editions,
                "previewUrl": starlyMetadata.previewUrl,
                "creatorName": starlyMetadata.collection.creator.name,
                "creatorUsername": starlyMetadata.collection.creator.username
            }
        )
    }
}